lib/node/nodes/georeference-street-address.js

```diff
@@ -49,42 +53,55 @@ GeoreferenceStreetAddress.prototype.getGeocoderParams = function () {
     return geocoderParams;
 };
 
-var COLUMNS_TEMPLATE_REGEX = /\{\{([^\}]*)\}\}/g;
+var COLUMNS_TEMPLATE_REGEX = /\{\{([^\}]*)\}\}/;
 GeoreferenceStreetAddress.prototype.getStreetAddressColumnParam = function() {
     if (this.street_address_column) {
         return this.getFunctionParam('street_address');
     }
 
-    var columns = [];
-    var result = this.street_address_template.replace(COLUMNS_TEMPLATE_REGEX, function(s, column) {
-        columns.push(column.trim());
-        return '%s';
+    var columns_and_literals = this.street_address_template.trim().split(', ').map(function(fragment) {
+        fragment = fragment.trim();
+        if(COLUMNS_TEMPLATE_REGEX.test(fragment)) {
+            var column_match = COLUMNS_TEMPLATE_REGEX.exec(fragment);
+            return column_match[1].trim();
+        } else {
+            return '\'\'' + fragment + '\'\'';
+        }
     });
 
-    if (columns.length === 0) {
-        return '$tpl$' + this.street_address_template + '$tpl$';
+    if (columns_and_literals.length === 0) {
+        return '$tpl$\'' + this.street_address_template + '\'$tpl$';
     }
 
-    return 'format($tpl$'+result+'$tpl$, ' + columns.join(', ') + ')';
+    return '$tpl$' + columns_and_literals.join(' || \'\', \'\' || ') + '$tpl$';
```

Comment:
This produces outputs like `$tpl$''Logroño'' || '', '' || ''La Rioja'' || '', '' || ''Spain''$tpl$`. Is that intentional for some reason I'm not understading in this context?

Reply:
Yes. it's used in the template, at `street_column`:

https://github.com/CartoDB/camshaft/pull/367/files#diff-6a7ad66626fc4d94638530f35f0ba1abR9. In that usage it's not SQL code yet: it's a string that will be executed by the extension. That string should be a valid SQL fragment.